Export stats primitives from Rust to CRuby

maximecb opened the pull request:
@XrXr would appreciate your input on whether I'm doing this right/wrong

I wanted the stats `Primitive` functions that are called from `yjit.rb` to be exported from Rust. I had to manually declare function prototypes in `yjit.h`, which we somehow didn't need to do before. I renamed the functions so they have `rb_yjit_` in the name for consistency.

On a sidenote, `RubyVM::YJIT` is not available.  Is calling `mYjit = rb_define_module_under(rb_cRubyVM, "YJIT");` sufficient to make `RubyVM::YJIT` available?

maximecb committed:
Export stats primitives from Rust to CRuby

XrXr committed:
Run yjit.rb during init again and move primitive functions from yjit.h

I removed code to run yjit.rb during init from `inits.c` earlier as an
expedient way to get builds working. It's led to confusion about why
RubyVM::YJIT isn't showing up. Sorry!

As for the functions added to yjit.h, I moved them to yjit_iface.c
because I think we want to keep yjit.h a shortlist of declaration for
code external to YJIT. The primitive functions only need to be visible
before the `yjit.rbinc` include.

I hope you don't mind me pushing to your branch. Feel free to tell me to
stop it.

XrXr committed:
preemptive diff minimization

XrXr commented:
I got `./miniruby -e 'p RubyVM::YJIT' printing and pushed it as` [7137d77](https://github.com/Shopify/ruby/pull/196/commits/7137d770785e9be330b1c1afbc102451bb5a9f4b). Please take a look :)

XrXr committed:
Remove unnecessary stubs

maximecb commented:
Thanks Alan.

The build is currently failing tho :O

XrXr committed:
Try to fix !YJIT_BUILD builds

too lazy to spin up a separate build config locally so sorry if this
fails CI again 😅

XrXr committed:
Fix again

XrXr committed:
Stub Init_builtin_yjit() in vm.o rather than in header

maximecb commented:
Thanks for your help in getting this working Alan. Hope you have a good weekend :)
